File: my-folder/1319-unique-number-of-occurrences/solution.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        numFreq = {}

        for num in arr:
            if num in numFreq.keys():
                numFreq[num] += 1
            else:
                numFreq[num] = 1

        UniqueOcc = set(numFreq.values())
        
        return len(numFreq) == len(UniqueOcc)

File: my-folder/1319-unique-number-of-occurrences/solution.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function(arr) {
    // let freq = {};

    // for(let i = 0; i < arr.length; i++) {
    //     if(!(arr[i] in freq)) {
    //         freq[arr[i]] = 1;
    //     } else {
    //         freq[arr[i]] += 1;
    //     }
    // }

    // const valueSet = new Set();
    // for(let value of Object.values(freq)) {
    //     if(valueSet.has(value)) {
    //         return false;
    //     }
    //     valueSet.add(value);
    // }
    // return true;
    
    let freq = new Map();

    arr.forEach(num => freq.set(num, freq.get(num) + 1 || 1));

    const set = new Set(freq.values());

    return set.size === freq.size;
};
